### PR TITLE
fix(f2v): Files list loading with special chars in uid

### PR DESCRIPTION
**Pull request by Pytal**
## Summary

Before | After
--- | ---
image | image

- Create dav client with correctly encoded root url `…/dav/files/test user/` → `…/dav/files/test%20user/`

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by AndyScherzinger — APPROVED** [review]

**Review by artonge — APPROVED** [review]

**Review by skjnldsv — CHANGES_REQUESTED** [review]

File: apps/files/src/services/WebdavClient.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ import { generateRemoteUrl } from '@nextcloud/router'
 import { getCurrentUser, getRequestToken } from '@nextcloud/auth'
 import { request } from 'webdav/dist/node/request.js'
 
-export const rootPath = `/files/${getCurrentUser()?.uid}`
+export const rootPath = `/files/${encodeURIComponent(getCurrentUser()?.uid || '')}`
```
[the remark this answers is not in the document]
Comment (Pytal) [reply apps/files/src/services/WebdavClient.ts]: https://github.com/nextcloud-libraries/nextcloud-files/pull/759

**Comment by Pytal** [comment]
Now correctly encodes single quotes as defined in https://github.com/nextcloud/server/blob/eb5510188ecc721bf43ad615b22d99bd98b46ee3/lib/private/User/Manager.php#L706

File: apps/files/src/services/WebdavClient.ts [hunk 2]
```diff
@@ -30,7 +30,10 @@ export const rootPath = `/files/${getCurrentUser()?.uid}`
 export const defaultRootUrl = generateRemoteUrl('dav' + rootPath)
 
 export const getClient = (rootUrl = defaultRootUrl) => {
-	const client = createClient(rootUrl, {
+	// Encode valid user id characters
+	const encodedUrl = encodeURI(rootUrl)
+		.replaceAll('\'', '%27')
```
Comment (skjnldsv) [inline apps/files/src/services/WebdavClient.ts:35]: We should not hae to replace quotes, they are allowed in an URL according to [rfc2396](https://www.ietf.org/rfc/rfc2396.txt)

I think the webdav have an issue:
Here are the exact two same breakpoints:
| Without quote in userId | With quote in userId |
|:---------:|:------:|
||image|

**Comment by Pytal** [comment]
Superseded by https://github.com/perry-mitchell/webdav-client/pull/360

**Pull request closed without merging**